src/modules/flow/timestamp/timestamp.c

```diff
@@ -0,0 +1,482 @@
+/*
+ * This file is part of the Soletta Project
+ *
+ * Copyright (C) 2015 Intel Corporation. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in
+ *     the documentation and/or other materials provided with the
+ *     distribution.
+ *   * Neither the name of Intel Corporation nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "timestamp-gen.h"
+#include "sol-flow-internal.h"
+
+#include <sol-util.h>
+#include <errno.h>
+#include <time.h>
+
+
+static int
+time_process(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id, const struct sol_flow_packet *packet)
+{
+    time_t current_time;
+
+    if (time(&current_time) == -1) {
+        sol_flow_send_error_packet(node, errno,
+            "Could not fetch current time: %s", sol_util_strerrora(errno));
+        return 0;
+    }
+
+    return sol_flow_send_timestamp_packet(node,
+        SOL_FLOW_NODE_TYPE_TIMESTAMP_TIME__OUT__OUT, &current_time);
+}
+
+struct make_time_data {
+    struct tm received_time;
+    int16_t initialized;
+};
+
+#define ALL_PORTS_INITIALIZED (63)
+
+static int
+send_timestamp(struct sol_flow_node *node, uint16_t port, struct make_time_data *mdata)
+{
+    time_t timestamp;
+
+    mdata->initialized |= 1 << port;
+
+    if (mdata->initialized != ALL_PORTS_INITIALIZED)
+        return 0;
+
+    timestamp = mktime(&mdata->received_time);
+    if (timestamp < 0) {
+        SOL_WRN("Failed to convert to timestamp");
```

Comment:
If one tries to set a time stamp from a date such as "32/01/2015", an error packet will be sent since a month can't have more than 31 days. However, if day is greater than the amount of days in a given month (e.g. 31 for months with only 30 days, or for days in February, where the rule changes on leap years, etc), only a `SOL_WRN` will be issued; no error packets will be sent.

Is this intentional?